packages/discovery/src/discovery/provider/ProviderWithCache.ts

```diff
@@ -165,7 +234,11 @@ export class ProviderWithCache extends DiscoveryProvider {
   override async getMetadata(
     address: EthereumAddress,
   ): Promise<ContractMetadata> {
-    const key = this.buildKey('getMetadata', [address])
+    const key = this.buildIdentity({
+      invocation: 'getMetadata',
+      params: [address],
+    })
```

Comment:
Hmm, unfortunately, this can change in case of a reorg...

Reply:
Honestly, when I think more about it, this is problematic only in one case and this is super narrow: we have a long-running process (ex. `UpdateMonitor`, `LZ Dashboard`) and a new contracts appears (is set in the `contractValues` of some contract) AND afterwards, it's deployment is reorged out of the chain. This is very slim edge case, BUT also something that would probably want to discover, as it could break the system quite a bit, 'cause we can assume it would break the system completely.

Reply:
After our talk, I agree that we can ignore those 2 calls are going to etherscan, which probably has some reorg-protections implemented. WYT @adamiak?